src/decouplededitor.js

```diff
@@ -57,23 +55,51 @@ export default class DecoupledEditor extends Editor {
 	 * {@link module:editor-decoupled/decouplededitor~DecoupledEditor.create `DecoupledEditor.create()`} method instead.
 	 *
 	 * @protected
-	 * @param {HTMLElement|String} sourceElementOrData The DOM element that will be the source for the created editor
+	 * @param {Object.<String,String>|Object.<String,HTMLElement>} sourceElementsOrData The object where each key
+	 * is the name of the root and corresponding value is the DOM element that will be the source for the created editor
 	 * (on which the editor will be initialized) or initial data for the editor. For more information see
-	 * {@link module:editor-balloon/ballooneditor~BalloonEditor.create `BalloonEditor.create()`}.
+	 * {@link module:editor-decoupled/decouplededitor~DecoupledEditor.create `DecoupledEditor.create()`}.
 	 * @param {module:core/editor/editorconfig~EditorConfig} config The editor configuration.
 	 */
-	constructor( sourceElementOrData, config ) {
+	constructor( sourceElementsOrData, config ) {
 		super( config );
 
-		if ( isElement( sourceElementOrData ) ) {
-			this.sourceElement = sourceElementOrData;
-		}
+		/**
+		 * The array of all sourceElements (which becomes editable areas) if any were passed in `sourceElementsOrData`.
+		 *
+		 * @protected
+		 * @member {Array.<HTMLElement>}
+		 */
+		this._sourceElements = [];
+
+		/**
+		 * The array of all root names.
+		 *
+		 * @protected
+		 * @member {Array.<String>}
+		 */
+		this._rootNames = sourceElementsOrData ? Object.keys( sourceElementsOrData ) : [];
 
 		this.data.processor = new HtmlDataProcessor();
 
-		this.model.document.createRoot();
+		const editables = [];
+
+		for ( const name of this._rootNames ) {
```

Comment:
That's a detail but `name` variable is meaningless for me. `rootName` is much better.